pg_search/src/index/directory/utils.rs

```diff
@@ -337,95 +337,118 @@ pub unsafe fn load_metas(
+                    // and it's the only one we want
+                    alive_segments.clear();
+                    alive_entries.clear();
+                    pin_cushion.clear();
+                } else {
+                    // we already have the largest so we don't need this entry
+                    need_entry = false;
+                }
+            }
 
-            opstamp = opstamp.max(Some(entry.opstamp()));
-        }
-    });
+            if need_entry {
+                pin_cushion.push(bman, &entry);
+                let inner_segment_meta = InnerSegmentMeta {
+                    max_doc: entry.max_doc,
+                    segment_id: entry.segment_id,
+                    deletes: entry.delete.map(|delete_entry| DeleteMeta {
+                        num_deleted_docs: delete_entry.num_deleted_docs,
+                        opstamp: 0, // hardcode zero as the entry's opstamp as it's not used
+                    }),
+                    include_temp_doc_store: Arc::new(AtomicBool::new(false)),
+                };
 
-    match solve_mvcc {
-        MvccSatisfies::ParallelWorker(only_these) if alive_entries.len() != only_these.len() => {
-            let missing = only_these
-                .difference(&alive_entries.iter().map(|s| s.segment_id).collect())
-                .cloned()
-                .collect::<HashSet<SegmentId>>();
-            let found = only_these.difference(&missing).collect::<HashSet<_>>();
+                alive_segments.push(inner_segment_meta.track(inventory));
+                alive_entries.push(entry);
 
-            panic!(
-                "load_metas: MvccSatisfies::ParallelWorker didn't load the correct segments. \
+                opstamp = opstamp.max(Some(entry.opstamp()));
+            }
+        });
+
+        match solve_mvcc {
+            MvccSatisfies::ParallelWorker(only_these)
+                if alive_entries.len() != only_these.len() =>
+            {
+                // If we haven't tried the `segment_metas_garbage` list, try that next.
+                if !exhausted_metas_lists {
+                    if let Some(garbage) = MetaPage::open(indexrel).segment_metas_garbage() {
+                        segment_metas = garbage;
+                        exhausted_metas_lists = true;
+                        continue;
+                    }
+                }
+
+                // TODO:  I believe this situation, where if the alive_entries.len() != only_these.len() is now dead code
+                //        @Stu asked to keep this around for awhile.  If there's ever a situation where `alive_entries` has
+                //        fewer entries than `only_these` then we should cancel the query, but it'd be indicative of a bug
+                //
+                // TODO (after some testing):  This situation does indeed happen and I believe that points to a bug, but
+                //        I don't know where it's coming from.  As such, cancelling the query is the expedient decision.
```

Comment:
If you need another CI cycle anyway, could replace this with an explanation of the situation:
> The parallel worker leader and parallel workers open the SegmentMetaEntrys list at different times, and
> in the presence of physical replication, cannot use pins or locks to prevent it from changing between
> those calls
>
> Instead, we ensure that the entries which the leader chose will still exist _somewhere_: a "garbage" list.

Reply:
I am not sure that anything I write here would be 100% accurate at the moment.  😬 

Lets discuss next week about if this is going to be our permanent solution and see how it works in prod.